src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -393,6 +393,29 @@ define('diagramStore', ['jquery', 'draw.io', 'xwiki-events-bridge'], function($)
   // We need to update the form fields before the form is validated (for Preview, Save and Save &amp; Continue).
   $(document).on('xwiki:actions:beforePreview xwiki:actions:beforeSave', updateFormFields);
 
+  var saveAttachment = function(file) {
+    file.getUi().exportToCanvas(function(canvas) {
+      if(canvas) {
```

Comment:
```suggestion
      if (canvas) {
```
I would also move the code within the IF in a separate ``saveCanvasAsAttachment`` function.